### PR TITLE
Load users and privileges from SQLServer

### LINKED ISSUE #9855: Load users and privileges from SQLServer
Hi,

This issue is part of #8966  and to load the privileges from Oracle. You can refer to `MySQLPrivilegeLoader` to finish this one. BTW, please give a look at #9851 for a better implement.

### Task
- Implement `SQLServer PrivilegeLoader `
- Add a test class for it (#9852 will give some hints).

Have a good day. :)

### PR DESCRIPTION
**Pull request by totalo**
Fixes #9855.

Changes proposed in this pull request:
- Load users and privileges from SQLServer

### CODE REVIEW

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/loader/storage/impl/dialect/SQLServerPrivilegeLoader.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * SQLServer privilege loader.
+ */
+public final class SQLServerPrivilegeLoader implements StoragePrivilegeLoader {
+
+    private static final String TABLE_PRIVILEGE_SQL =
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/loader/storage/impl/dialect/SQLServerPrivilegeLoader.java:45]: So is there any privilege having nothing to do with `Table`? E.g., `BACKUP DATABASE,`BACKUP LOG`, `CREATE DATABASE`?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/loader/storage/impl/dialect/SQLServerPrivilegeLoader.java (hunk 1 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/loader/storage/impl/dialect/SQLServerPrivilegeLoader.java:45]: Yes, it has, I didn’t find the relevant information at first, now I have added it

**Comment by tristaZero** [comment]
@totalo Sorry that there are some conflict files to fix. But take it easy, since it is about package adjustments, it is not hard to fix.

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @totalo ,

I know the privilege system of SQLServer is large and complicated, but we can firstly fill some necessary privileges of `Server level`, `DB` and `table`.

[principals-database-engine](https://docs.microsoft.com/en-us/sql/relational-databases/security/authentication-access/principals-database-engine?view=sql-server-ver15) and[Guide to SQL Server Permissions](https://blog.netwrix.com/2020/10/16/guide-to-sql-server-permissions/)  give us a brief introduction on SQLserver's privileges. We can learn that there are two layers of principals, i.e., `Server-level` and `Database-level`. 

I assume that 
- `Server-level` privileges is bound to be mapped to our `AdministrativePrivileges` and `DatabasePrivileges`
- `DB-lever` privileges belong to `globalPrivileges` of `DatabasePrivileges` or `specificPrivileges`
- Still we have table some privileges putting into `TablePrivileges`

Also, there are many different specific privileges on `server-level` and `db-lever`, right? You can pick up the main ones as the first step.

This is just my opinion, welcome your any reply.

**Comment by totalo** [comment]
> [principals-database-engine](https://docs.microsoft.com/en-us/sql/relational-databases/security/authentication-access/principals-database-engine?view=sql-server-ver15)

Yes, I think so too. I have already looked at principals-database-engine when searching for information. Thank you for providing me with these three directions, please give me some time, I will study it, and then modify this pr

**Comment by totalo** [comment]
> Hi @totalo ,
> 
> I know the privilege system of SQLServer is large and complicated, but we can firstly fill some necessary privileges of `Server level`, `DB` and `table`.
> 
> [principals-database-engine](https://docs.microsoft.com/en-us/sql/relational-databases/security/authentication-access/principals-database-engine?view=sql-server-ver15) and[Guide to SQL Server Permissions](https://blog.netwrix.com/2020/10/16/guide-to-sql-server-permissions/) give us a brief introduction on SQLserver's privileges. We can learn that there are two layers of principals, i.e., `Server-level` and `Database-level`.
> 
> I assume that
> 
> * `Server-level` privileges is bound to be mapped to our `AdministrativePrivileges` and `DatabasePrivileges`
> * `DB-lever` privileges belong to `globalPrivileges` of `DatabasePrivileges` or `specificPrivileges`
> * Still we have table some privileges putting into `TablePrivileges`
> 
> Also, there are many different specific privileges on `server-level` and `db-lever`, right? You can pick up the main ones as the first step.
> 
> This is just my opinion, welcome your any reply.

Yes, I think so too. I have already looked at principals-database-engine when searching for information. But at the time, my grasp of the direction was not very clear. Thank you for providing me with these three directions, please give me some time, I will study it, and then revise this PR.

**Comment by tristaZero** [comment]
@totalo Take it easy, there are plenty of time for you to learn more and make this PR perfect.

**Comment by totalo** [comment]
I will submit again when the modification is completed

**Comment by totalo** [comment]
> @totalo Take it easy, there are plenty of time for you to learn more and make this PR perfect.

Hello, I tried to fix a version, please review it.

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-api/src/main/java/org/apache/shardingsphere/authority/model/ShardingSpherePrivileges.java [hunk 2]
```diff
@@ -23,28 +23,28 @@
  * ShardingSphere Privileges.
  */
 public interface ShardingSpherePrivileges {
-    
+
     /**
      * Set super privilege.
      */
     void setSuperPrivilege();
-
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-authority/shardingsphere-authority-api/src/main/java/org/apache/shardingsphere/authority/model/ShardingSpherePrivileges.java:31]: Hi @totalo,

Thanks for your efforts. It looks much better.
Just one issue here.  There is no necessary changes in this file, I guess,  could you  revert it?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-api/src/main/java/org/apache/shardingsphere/authority/model/ShardingSpherePrivileges.java (hunk 2 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-api/src/main/java/org/apache/shardingsphere/authority/model/ShardingSpherePrivileges.java:31]: ok~

**Review by tristaZero — APPROVED** [review]
Hi @totalo Thanks for your nice work. :)

**Pull request merged**